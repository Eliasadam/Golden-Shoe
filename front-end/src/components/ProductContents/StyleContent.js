import styled from 'styled-components';

export const CardWrapper = styled.div`
  display: block;
  flex: 1;
  box-shadow: 0 4px 6px #0d506a;
  background-color: #f3feef;
  align-items: center;
  justify-content: center;
  height: 260px;
  width: 220px;
  margin: 1rem;
  padding: 0.5rem;
`;
export const CardList = styled.div`
  display: flex;
  flex-direction: row;
  align-items: center;
  width: 100%;
  flex-wrap: wrap;
  justify-content: space-between;
  margin: 0.5rem;
`;
export const CardTitle = styled.p`
  font-size: 12px;
  align-self: center;
  margin: 1rem;
  text-align: center;
`;
export const CardImg = styled.img`
  width: 150px;
  height: 150px;
  margin: 0.5rem;
  border-radius: 0.2rem;
`;