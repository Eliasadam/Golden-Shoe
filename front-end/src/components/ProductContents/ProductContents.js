import React, { useContext } from 'react';
import { ContentList, ProductImg, ProductTitle } from './StyleContent';
import { GlobalContext } from '../../Contexts/GlobalState';

const ProductContents = () => {
  const { query } = useContext(GlobalContext);
  return (
    <div data-testid="card">
      {query? query.map((product, index) => {
        const uniqueKey = `${product}${index}`;
        return (
          <ContentList key={uniqueKey} >
            <ProductImg src={product.src} />
            <ProductTitle>{product.title}</ProductTitle>
            <ProductTitle> Colour: {product.colors.length}</ProductTitle>
            <ProductTitle>Â£{product.price}</ProductTitle>
          </ContentList>
        );
      }): []}
    </div>
  );
};

export default ProductContents;
