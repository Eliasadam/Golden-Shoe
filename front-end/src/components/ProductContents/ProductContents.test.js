import React from 'react';
import { render, cleanup } from '@testing-library/react';
import ProductContents from './ProductContents';

afterEach(cleanup);
const contents = {
    src: "https://www.upsieutoc.com/images/2020/06/27/img1.jpg",
    content: "nike balance",
    price: 23,
    colors: ["red", "black", "crimson", "teal"]
};

describe('This suit is to test the ProductContents component', () => {
    test('Snapshot of ProductContents', () => {
      const { asFragment } = render(<ProductContents />);
      expect(asFragment()).toMatchSnapshot();
    });
    test("finding title with TestId", () => {
        const { getByTestId } = render(<ProductContents contents={contents}/>);
        const card = getByTestId("card");
      expect(card).toBeDefined();
    });
  });
  