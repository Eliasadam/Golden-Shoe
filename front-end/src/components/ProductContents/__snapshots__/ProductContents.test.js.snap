// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`This suit is to test the ProductContents component Snapshot of ProductContents 1`] = `
<DocumentFragment>
  <div
    data-testid="card"
  />
</DocumentFragment>
`;
